src/lib/utils.js

```diff
@@ -102,7 +102,9 @@ const whitelistURICodes = whitelistURI.split('').map(c => encodeURIComponent(c))
 const whitelistRegExp = new RegExp(`(?:${whitelistURICodes.join('|')})`, 'g');
 
 /**
- * Encode all invalid URI characters, except the ones we've decided we don't want to
+ * Encodes all invalid URI characters. Used to apply encodeURIComponent on an entire query string at
+ * once, without having to separate out the parameters, as it skips the characters that shouldn't
+ * be encoded.
  */
 export function customEncodeURIComponent(uri) {
```

Comment:
I'm not really sure why this function exists. Seems like you could just use encodeURI instead of this? This custom encoder encodes a little bit more than encodeURI, but I don't understand why these specific characters. Seems to me that all you'd need to skip are `?=&` to safely use it to encode an entire query string at once?

Reply:
I think yes, those whitelisted characters are used in the Web API for metadata object and field filtering.
See https://docs.dhis2.org/master/en/developer/html/webapi_metadata_object_filter.html and https://docs.dhis2.org/master/en/developer/html/webapi_metadata_field_filter.html.
That said, I'm not sure it's a problem to have them encoded normally via `encodeURIComponent`. It could be that is done that way to make it more readable and usable in dev tools, the URL can be copy pasted and modified easily without all the commas encoded (for example).

Reply:
Yeah, I wasn't 100% sure either. I saw that `customEncodeURIComponent` is used in another place as well, and at that point I thought that it would probably be better to tackle this in a separate PR.